tests/test_elns.py

```diff
@@ -0,0 +1,115 @@
+def test_connect_to_eln(mock_eln_config):
+    """Test the connect_to_eln function."""
+    from aiidalab_eln import CheminfoElnConnector
+
+    from aiidalab_widgets_base.elns import ELN_CONFIG, connect_to_eln
+
+    mock_eln_config.mock(original_config=ELN_CONFIG)
+
+    eln, message = connect_to_eln(eln_instance="any_random_eln")
+    assert eln is None
+    assert message.startswith("Can't open")
+
+    # Now, create a config file with {} content
+    mock_eln_config.write(config_dictionary={})
+
+    # Now check that empty config file is identified, but has no content.
+    eln, message = connect_to_eln()
+    assert eln is None
+    assert message.startswith(
+        "No ELN instance was provided, the default ELN instance is not configured either."
+    )
+    eln, message = connect_to_eln(eln_instance="any_random_eln")
+    assert eln is None
+    assert message == "Didn't find configuration for the 'any_random_eln' instance."
+
+    # Populate the config file with some content.
+    mock_eln_config.populate_mock_config_with_cheminfo()
+
+    # Now check that the config file is identified and has content.
+    eln, message = connect_to_eln()
+    assert isinstance(eln, CheminfoElnConnector)
+
+    mock_eln_config.restore()
+
+
+def test_eln_import_widget():
+    """Test the ElnImportWidget."""
+
+    # At the moment, we do a very minimal test of the widget, because it requires a running ELN instance.
+    # TODO: add a mock ELN instance.
+
+    from aiidalab_widgets_base.elns import ElnImportWidget
+
+    widget = ElnImportWidget()
+    assert widget.node is None
+
+
+def test_eln_export_widget(structure_data_object, mock_eln_config):
+    """Test the ElnExportWidget."""
+
+    # At the moment, we do a very minimal test of the widget, because it requires a running ELN instance.
+    # TODO: add a mock ELN instance.
```

Comment:
```suggestion
```
This is done by mock eln config, I assume?

Reply:
Not really. The config is mocked, but the sending and receiving information to/from ELN is not mocked.